R/select.R

```diff
@@ -0,0 +1,74 @@
+
+select_pos <- function(.x,
+                       ...,
+                       .include = NULL,
+                       .exclude = NULL,
+                       .strict = TRUE) {
+  vctrs::vec_assert(.x)
+
+  vars <- names(.x)
+  if (is_null(vars)) {
+    abort("Can't select within an unnamed vector.")
+  }
+
+  scoped_vars(vars)
+  select_impl(
+    .x,
+    ...,
+    .include = .include,
+    .exclude = .exclude,
+    .strict = .strict
+  )
+}
+
+# Example of implementation, mainly used for unit tests
+vec_select <- function(.x, ..., .strict = TRUE) {
+  if (is.data.frame(.x)) {
+    abort("`.x` can't be a data frame.")
+  }
+  idx <- select_pos(.x, ..., .strict = .strict)
+  set_names(vctrs::vec_slice(.x, idx), names(idx))
+}
+
+select_impl <- function(.x = NULL,
+                        ...,
+                        .include = NULL,
+                        .exclude = NULL,
+                        .strict = TRUE) {
+  dots <- enquos(...)
```

Comment:
Would it be worth making this `vars <- c(!!!enquos(...))`? It would simplify the `.include` and `.exclude` implementation and would reduce possible inconsistencies.

Reply:
Yup this is where we'll end up once `c(...)` is consistent with current `...` behaviour. I need to fix input joining, and duplicate name checking.

Reply:
Ok, cool